examples/src/test/scala/docs/cassandra/CassandraProjectionDocExample.scala

```diff
@@ -124,6 +125,30 @@ object CassandraProjectionDocExample {
     //#grouped
   }
 
+  object IllustrateAtLeastOnceFlow {
+    import akka.persistence.query.Offset
+    //#atLeastOnceFlow
+    val logger = LoggerFactory.getLogger(getClass)
+
+    val flow = FlowWithContext[EventEnvelope[ShoppingCart.Event], EventEnvelope[ShoppingCart.Event]]
+      .map(envelope => envelope.event)
+      .map {
+        case ShoppingCart.CheckedOut(cartId, time) =>
+          logger.info("Shopping cart {} was checked out at {}", cartId, time)
+          Done
+
+        case otherEvent =>
+          logger.debug("Shopping cart {} changed by {}", otherEvent.cartId, otherEvent)
+          Done
+      }
```

Comment:
I understand the intention to keep the documentation simple, but I found it confusing that we are showing an example that only does side-effect while the real implementation that users are interested is the `mapAsync`. 

We could replace this by something like:

```scala
FlowWithContext[Envelope, Envelope]
  .mapAsync(1) { 
    case ShoppingCart.CheckedOut(cartId, time) =>
 	  logger.info("Shopping cart {} was checked out at {}", cartId, time)
      // process the event using an async repository
      repository.saveCartCheckoutTime(cartId, time)
}
```

then we could add in text that `repository` is processing the 'save' asynchronously, etc..

Reply:
I agree that we should come up with a better example.
`mapAsync(1)` isn't good either, because that could more easily be done in a handler.

Reply:
What would be then the most idiomatic way of using it considering that at the end we should have a `Future[Done]`?